source/developers/experience-builder.rst

```diff
@@ -1139,65 +1208,468 @@ breadcrumb
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The easiest way to integrate XB with your JS application is by putting attributes on each HTML element that
+represents a model, field or item of a CrafterCMS content type and then invoking XB initializer.
+
+.. _fetchIsAuthoring:
+
+fetchIsAuthoring
+""""""""""""""""
+
+This function checks against the specified CrafterCMS server if it is running against an authoring server.
+When running in authoring, in-context editing tools should be enabled in the application whilst in
+delivery (i.e. "production"), they should not.
+
+The function returns a promise which will resolve as true or false. This value should be fetch early
+on your application bootstrap and cached for the rest of the app lifecycle. Depending on the value,
+you should then carry on to initialize XB or bypass it's initialization and assume the app is running
+in "production", where authoring tools are completely absent.
+
+.. TODO Internally it uses `crafterConf < add docs on readme and link to them >`_
+
+.. code-block:: js
+
+      import { fetchIsAuthoring, initExperienceBuilder } from '@craftercms/experience-builder';
+
+      // Check if we're in authoring
+      fetchIsAuthoring().then((isAuthoring) => {
+         // If we're in authoring, initialize XB
+         if (isAuthoring) {
+            initExperienceBuilder()
+         }
+      })
+
+.. list-table::
+   :widths: 10 10 10 70
+   :header-rows: 1
+
+   * - Parameter
+     - Type
+     - Default
+     - Description
+   * - ``config``
+     - Record<'baseUrl' | 'site', string>
+     - undefined
+     - You can supply a baseUrl and/or site to make the check. ``fetchIsAuthoring`` uses ``crafterConf``
+       (from ``@craftercms/classes`` package) values when not supplied.
 
 .. TODO
+      Is addAuthoringSupport still needed? If used via npm, everything is imported from the package and,
+      if imported as a script, everything is already loaded.
+
+      addAuthoringSupport
+      """""""""""""""""""
+
+      Add authoring support will import the XB scripts on to your page.
+
+.. _getICEAttributes:
 
-        ^^^^^
-        React
-        ^^^^^
+getICEAttributes
+""""""""""""""""
+
+Use this method to get the set of attributes to place on each element that represents a CrafterCMS
+model, field or item. Once you've fetched your content, you'd invoke ``getICEAttributes`` and it will
+return all the necessary attributes to inform the system how to make such element editable in XB.
+
+You should first set all the attributes on your markup and afterwards, invoke `initExperienceBuilder <#initexperiencebuilder>`_
+
+.. list-table::
+   :widths: 10 10 10 70
+   :header-rows: 1
 
-        CrafterCMS provides react bindings for integrating with XB.
+   * - Parameter
+     - Type
+     - Default
+     - Description
+   * - ``config``
+     - `ICEConfig <https://github.com/craftercms/studio-ui/blob/33b003c49fdde3ea00e1d95ca02d9f1e6869b301/ui/guest/src/index.tsx#L40>`_
```

Comment:
Should we make it point to the develop branch instead?
`https://github.com/craftercms/studio-ui/blob/develop/ui/guest/src/index.tsx#L40`